wallet-test/src/test/scala/org/bitcoins/wallet/UTXOLifeCycleTest.scala

```diff
@@ -70,10 +149,49 @@ class UTXOLifeCycleTest extends BitcoinSWalletTest {
       // Need to call this to actually update the state, normally a node callback would do this
       _ <- wallet.updateUtxoPendingStates()
       confirmedCoins <- wallet.spendingInfoDAO.findOutputsBeingSpent(tx)
-    } yield assert(confirmedCoins.forall(_.state == ConfirmedSpent))
+
+      // Assert tx is confirmed
+      _ = assert(confirmedCoins.forall(_.state == ConfirmedSpent))
+      _ = assert(confirmedCoins.forall(_.spendingTxIdOpt.contains(tx.txIdBE)))
+
+      // Create tx to spend same utxos
+      newTx = {
+        val inputs = updatedCoins.map { db =>
+          TransactionInput(db.outPoint, EmptyScriptSignature, UInt32.zero)
+        }
+        BaseTransaction(Int32.zero, inputs, Vector.empty, UInt32.zero)
+      }
+
+      res <- recoverToSucceededIf[RuntimeException](
```

Comment:
If I understand this, we are not asserting our current expected behavior which we talked about here. The new spending txid should be in the database rather than the old one, right?

https://github.com/bitcoin-s/bitcoin-s/pull/2744#discussion_r587752135

Specifically 

>Do we have unit tests for an old spending transaction being overwritten by a new spending transaction? If so could you link me?

https://github.com/bitcoin-s/bitcoin-s/pull/2744/files#diff-fd826a398083437abb5cf5acafa8af283c5420c626aa902469a5b743b8dfa094R285

Reply:
> If I understand this, we are not asserting our current expected behavior which we talked about here. The new spending txid should be in the database rather than the old one, right?

That is the case for pending conf utxos, this is asserting it fails when the utxos are confirmed

Reply:
> Do we have unit tests for an old spending transaction being overwritten by a new spending transaction? If so could you link me?

https://github.com/bitcoin-s/bitcoin-s/pull/2744/files#diff-18b1ff078f6c3b0ec3a3ee865e921754d7e4b7ec485fcb64b2d463704748d8aaR96